assets/src/edit-story/app/story/useStoryReducer/actions.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Internal dependencies
+ */
+import * as types from './types';
+
+const addPage = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.ADD_PAGE, payload: { properties } } );
+
+const deletePage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.DELETE_PAGE, payload: { pageIndex } } );
+
+const deleteCurrentPage = ( dispatch ) => () =>
+	dispatch( { type: types.DELETE_PAGE, payload: { pageIndex: null } } );
+
+const updatePageProperties = ( dispatch ) => ( { pageIndex, properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageIndex, properties } } );
+
+const updateCurrentPageProperties = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageIndex: null, properties } } );
+
+const duplicatePage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.DUPLICATE_PAGE, payload: { pageIndex } } );
+
+const duplicateCurrentPage = ( dispatch ) => () =>
+	dispatch( { type: types.DUPLICATE_PAGE, payload: { pageIndex: null } } );
+
+const movePage = ( dispatch ) => ( { pageIndex, position } ) =>
+	dispatch( { type: types.MOVE_PAGE, payload: { pageIndex, position } } );
+
+const setCurrentPage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.SET_CURRENT_PAGE, payload: { pageIndex } } );
+
+const addElements = ( dispatch ) => ( { elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex: null, elements } } );
+
+const addElement = ( dispatch ) => ( { element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex: null, elements: [ element ] } } );
+
+const addElementsToPage = ( dispatch ) => ( { pageIndex, elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex, elements } } );
+
+const addElementToPage = ( dispatch ) => ( { pageIndex, element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex, elements: [ element ] } } );
+
+const deleteElementsById = ( dispatch ) => ( { elementIds } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds } } );
+
+const deleteSelectedElements = ( dispatch ) => () =>
```

Comment:
I always wanted selection to be part of this, but not "editing element" (that's left for the canvas only). Selection is directly influenced by actions in the story state, e.g. adding a new page clears selection, adding a new element makes that element the selection, etc.

Reply:
Also, `selection` here refers specifically to the elements that have focus right now on the current page and is a list of their ids. Not the selected page (current page) nor the focused page, element in edit mode, etc.

Reply:
Sounds good. My only (smaller) concerns are:

1. There are two selection operations: toggle and set.
2. There could be some "addNew" or "duplicate" cases where the selection stays on the previous element.

I don't think we'll run into these right away, so it's probably ok for now. But I'm slightly concern that we will have a bunch of complicated "selectionOperation" flags, such as "add", "toggle", "set", "auto", "none".

Reply:
> 1. There are two selection operations: toggle and set.

If you click an element (on page or in layer panel), it and only it should be selected. If you lasso-select a bunch of elements on page, only those exact elements should be selected. If you shift-click from one element to another in the layer panel, those two elements and all elements in between should be selected. All these ways of selecting will use `setSelectedElements`.

If you shift-click an element, it will be toggled in the current selection and doesn't need to know which other elements are currently selected. This will use `toggleElementInSelection`.

Maybe `add` and `remove` will never be used, now that I think of it. If not, let's definitely remove them. They're very simple operations however.

> 2. There could be some "addNew" or "duplicate" cases where the selection stays on the previous element.

I've tried to think hard about what the selection state should be after all operations as specified in the jsdoc within each reducer - see for example [`addPage`](https://github.com/ampproject/amp-wp/pull/3910/files#diff-2140a3febf9327cf423699f615121608R13) or [`deleteElements`](https://github.com/ampproject/amp-wp/pull/3910/files#diff-8952b07de8f790551a5a26bed1b97b4fR16-R17). If you have specific concerns, please mention where appropriate.

Reply:
Ok. I think we won't figure it out fully in this PR. Let's go ahead and complete/merge and we'll be following up while implementing all of these features.